src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -127,35 +126,44 @@ private void DeterminePropertyName(JsonSerializerOptions options)
                 // Set the compare name.
                 if (options.PropertyNameCaseInsensitive)
                 {
-                    CompareNameAsString = NameAsString.ToUpperInvariant();
-                    _compareName = Encoding.UTF8.GetBytes(CompareNameAsString);
+                    NameUsedToCompareAsString = NameAsString.ToUpperInvariant();
+                    _nameUsedToCompare = Encoding.UTF8.GetBytes(NameUsedToCompareAsString);
                 }
                 else
                 {
-                    CompareNameAsString = NameAsString;
-                    _compareName = _name;
+                    NameUsedToCompareAsString = NameAsString;
+                    _nameUsedToCompare = _name;
                 }
 
                 // Cache the escaped name.
+                _escapedName = _name; // temporary behavior until the reader can accept escaped string.
+#if false
```

Comment:
I don't understand this. What reader feature is needed?

Reply:
Sorry that should be writer.